src/components/Header.js

```diff
@@ -8,54 +7,59 @@ import {
   NavItem,
   Collapse,
   Button,
-  Modal,
-  ModalHeader,
-  ModalBody,
-  FormGroup,
-  Label,
 } from 'reactstrap';
-import { Control, LocalForm, Errors } from 'react-redux-form';
-import { required, minLength, validEmail } from '../../validations/index';
+import { connect } from 'react-redux';
+import SignUser from './SignUser';
+import { postSignIn, postLogout } from '../stores/users/userActionCreators';
+
+const mapStateToProps = (state) => ({
+  user: state.user,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  postSignIn: (email, password) => dispatch(postSignIn(email, password)),
+  postLogout: (token) => dispatch(postLogout(token)),
+});
 
 class Header extends Component {
   constructor(props) {
     super(props);
     this.state = {
       isNavOpen: false,
       isModalOpen: false,
+      isLogin: false,
     };
+
     this.toggleNav = this.toggleNav.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
-    this.hanldeSubmit = this.hanldeSubmit.bind(this);
+    this.setLogin = this.setLogin.bind(this);
+    this.userLogout = this.userLogout.bind(this);
+  }
+
+  setLogin() {
+    this.setState({ isLogin: true });
+    this.setState((prevState) => ({
+      isModalOpen: !prevState.isModalOpen,
+    }));
+  }
```

Comment:
This can be done in only one `this.setState`, remember everything you change `state` you cause a re-render with React and now you are doing an extra one because having two setState